code-journal/js/main.js

```diff
@@ -149,21 +158,59 @@ document.addEventListener('DOMContentLoaded', function (event) {
   if (data.profile.username === '') {
     viewSwapping('edit-profile');
   } else {
-    viewSwapping('profile');
+    viewSwapping(data.view);
   }
 
 });
 
 // listen for link
 
 document.addEventListener('click', function (event) {
-  var $dataview = event.target.getAttribute('data-view');
+  data.view = event.target.getAttribute('data-view');
```

Comment:
You would want to do this in the else block where the user is able to navigate to their chosen view. In the block that navigates to `edit-profile`, you will want to set it equal to `edit-profile` 👍